### PR TITLE
Revert "Handle Jobs with ttl_seconds_after_finished = 0 correctly"

### PR DESCRIPTION
**Pull request by justinmchase**
Reverts hashicorp/terraform-provider-kubernetes#2596

Per issue #2649, please revert this breaking change which regresses a critical feature.

### CODE REVIEW

File: .changelog/2596.txt [hunk 1]
```diff
@@ -1,3 +0,0 @@
-```release-note:enhancement
-Properly handle Kubernetes Jobs with ttl_seconds_after_finished = 0 to prevent unnecessary recreation.
-```
```
Comment (arybolovlev) [inline .changelog/2596.txt:3]: Please don't remove this file. Instead, please add a new one that will reflect your changes. Your changes fall under the _bug fix_ category. Please mention there that this bug fix reverts changes introduced in 2.34.0.

**Review by arybolovlev — CHANGES_REQUESTED** [review]
Hi @justinmchase,

Thank you for reverting this change and properly reporting the related issue. Indeed, it looks like a breaking change and we need to find a better way to cover both scenarios without introducing a breaking change in the minor release.

I have left a small comment. Once it is addressed we can merge this change and cut a patch release.

File: .changelog/2596.txt (hunk 1 above)
Comment (justinmchase) [reply .changelog/2596.txt:3]: Makes sense, will do.

**Comment by justinmchase** [comment]
I updated the release notes files as requested and merged upstream main.

**Review by arybolovlev — APPROVED** [review]
Thanks, @justinmchase!

**Comment by dmpakhar** [comment]
This change broke functionality, when job should be created once and then deleted.
In my scenario k8s job should be launched only once. Now this job recreates every apply and there are no ability not to skip creation of this job in the second run or small changes that would not require execution of the job.

**Comment by justinmchase** [comment]
@dmpakhar Is it possible to add a separate field to get the behavior you want without regressing the behavior that is currently expected when ttl seconds after finish is 0?

If it helps, I will tell you that our Job is running a migration pod, the migration pod does run on every deployment but internally it is keeping track of the state of migration steps and applying them only once. This is a pretty standard way of handling run-once behavior rather than trying to manage it with terraform and kubernetes.

**Comment by dmpakhar** [comment]
@justinmchase 
Hi,
I think this can be managed as an extra flag. From my perspective this behaviour should be defined via extra var. this will make logic clear for everyone.

Now we went in same issue, when same parameter changes behaviour.

I think execute_once/recreate_after_finished = true/false variable may be a good idea

**Pull request merged**